src/core/lib/parse-directive-tag.ts

```diff
@@ -0,0 +1,259 @@
+/*
+* parse-directive-tag.ts
+*
+* Recognizes and parses directive opening, closing and self-closing tags.
+*
+* Copyright (C) 2022 by RStudio, PBC
+*
+*/
+
+// xml regex fragments
+//
+// this requires /u in the regex specifier !
+// https://www.w3.org/TR/2006/REC-xml11-20060816/#NT-NameStartChar
+const nameStartChar =
+  `[:A-Z_a-z\u{C0}-\u{D6}\u{D8}-\u{F6}\u{F8}-\u{2FF}\u{370}-\u{37D}\u{37F}-\u{1FFF}\u{200C}-\u{200D}\u{2070}-\u{218F}\u{2C00}-\u{2FEF}\u{3001}-\u{D7FF}\u{F900}-\u{FDCF}\u{FDF0}-\u{FFFD}\u{10000}-\u{EFFFF}]`;
+const nameChar =
+  `(?:${nameStartChar}|[-.0-9\u{B7}\u{0300}-\u{036F}\u{203F}-\u{2040}])`;
+const name = `(?:${nameStartChar}${nameChar}*)`;
+
+// https://www.w3.org/TR/xml/#NT-CharRef
+const entityRef = `[&]${name}[;]`;
+const charRef = `(?:[&][#][0-9]+[;]|[&][#]x[0-9a-fA-F]+[;])`;
+const reference = `(?:${entityRef}|${charRef})`;
+const attrValue =
+  `(?:["](?:[^<&"]|${reference})*["]|['](?:[^<&']|${reference})*['])`;
+
+const attribute = `(?:${name}\\s*=\\s*${attrValue})`;
+
+const htmlTagNames = new Set([
+  "a",
+  "abbr",
+  "acronym",
+  "address",
+  "applet",
+  "area",
+  "article",
+  "aside",
+  "audio",
+  "b",
+  "base",
+  "basefont",
+  "bdi",
+  "bdo",
+  "bgsound",
+  "big",
+  "blink",
+  "blockquote",
+  "body",
+  "br",
+  "button",
+  "canvas",
+  "caption",
+  "center",
+  "cite",
+  "code",
+  "col",
+  "colgroup",
+  "content",
+  "data",
+  "datalist",
+  "dd",
+  "del",
+  "details",
+  "dfn",
+  "dialog",
+  "dir",
+  "div",
+  "dl",
+  "dt",
+  "em",
+  "embed",
+  "fieldset",
+  "figcaption",
+  "figure",
+  "font",
+  "footer",
+  "form",
+  "frame",
+  "frameset",
+  "head",
+  "header",
+  "h1",
+  "hgroup",
+  "hr",
+  "html",
+  "i",
+  "iframe",
+  "image",
+  "img",
+  "input",
+  "ins",
+  "kbd",
+  "keygen",
+  "label",
+  "legend",
+  "li",
+  "link",
+  "main",
+  "map",
+  "mark",
+  "marquee",
+  "menu",
+  "menuitem",
+  "meta",
+  "meter",
+  "nav",
+  "nobr",
+  "noembed",
+  "noframes",
+  "noscript",
+  "object",
+  "ol",
+  "optgroup",
+  "option",
+  "output",
+  "p",
+  "param",
+  "picture",
+  "plaintext",
+  "portal",
+  "pre",
+  "progress",
+  "q",
+  "rb",
+  "rp",
+  "rt",
+  "rtc",
+  "ruby",
+  "s",
+  "samp",
+  "script",
+  "section",
+  "select",
+  "shadow",
+  "slot",
+  "small",
+  "source",
+  "spacer",
+  "span",
+  "strike",
+  "strong",
+  "style",
+  "sub",
+  "summary",
+  "sup",
+  "table",
+  "tbody",
+  "td",
+  "template",
+  "textarea",
+  "tfoot",
+  "th",
+  "thead",
+  "time",
+  "title",
+  "tr",
+  "track",
+  "tt",
+  "u",
+  "ul",
+  "var",
+  "video",
+  "wbr",
+  "xmp",
+]);
+
+export function isDirectiveTag(str: string) {
```

Comment:
If we don't match a directive, do we just assume it's an html custom tag (i.e. component)?  (seems like we should as if we don't that would preclude use of these tags)

Reply:
Yes, if it doesn't match a directive tag, then `breakQuartoMd` just adds the line to the markdown cell. And if the xml tag is well formed, then the only way `isDirectiveTag` returns true is if the tag name is an HTML element.